diff --git a/lib/-ftb-fzf b/lib/-ftb-fzf
index 62f93f1..03b7100 100755
--- a/lib/-ftb-fzf
+++ b/lib/-ftb-fzf
@@ -88,10 +88,8 @@ if [[ "$use_fzf_default_opts" == "yes" ]]; then
 fi
 
 FZF_DEFAULT_OPTS=$fzf_default_opts SHELL=$ZSH_NAME $fzf_command \
-  --ansi \
   --bind=$binds \
   --bind="${switch_group[1]}:reload($reload_command -1),${switch_group[2]}:reload($reload_command 1)" \
-  --color=hl:$(( header_lines == 0 ? 188 : 255 )) \
   --cycle \
   --delimiter='\x00' \
   --expect=$continuous_trigger,$print_query,$accept_line \
