diff --git a/lib/-ftb-fzf b/lib/-ftb-fzf
index 516d9fa..1f757f3 100755
--- a/lib/-ftb-fzf
+++ b/lib/-ftb-fzf
@@ -82,10 +82,8 @@ fi
 _ftb_query="${_ftb_query}$(command "$dd" bs=1G count=1 status=none iflag=nonblock < /dev/tty 2>/dev/null)" || true
 
 FZF_DEFAULT_OPTS='' SHELL=$ZSH_NAME $fzf_command \
-  --ansi \
   --bind=$binds \
   --bind="${switch_group[1]}:reload($reload_command -1),${switch_group[2]}:reload($reload_command 1)" \
-  --color=hl:$(( header_lines == 0 ? 188 : 255 )) \
   --cycle \
   --delimiter='\x00' \
   --expect=$continuous_trigger,$print_query,$accept_line \
