diff --git a/highlighters/main/main-highlighter.zsh b/highlighters/main/main-highlighter.zsh
index 3ec96d9..7de32d5 100644
--- a/highlighters/main/main-highlighter.zsh
+++ b/highlighters/main/main-highlighter.zsh
@@ -372,6 +372,8 @@ _zsh_highlight_highlighter_main_paint()
     'grc' :se # grc - a "generic colouriser" (that's their spelling, not mine)
     'cpulimit' elp:ivz # cpulimit 0.2
     'ktrace' fgpt:aBCcdiT
+    'yestty' ''
+    'notty' ''
   )
   # Commands that would need to skip one positional argument:
   #    flock
