diff --git a/highlighters/main/main-highlighter.zsh b/highlighters/main/main-highlighter.zsh
index e6f5e35..4ed391e 100644
--- a/highlighters/main/main-highlighter.zsh
+++ b/highlighters/main/main-highlighter.zsh
@@ -373,6 +373,8 @@ _zsh_highlight_highlighter_main_paint()
     'cpulimit' elp:ivz # cpulimit 0.2
     'ktrace' fgpt:aBCcdiT
     'caffeinate' tw:dimsu # as of macOS's caffeinate(8) dated November 9, 2012
+    'yestty' 'ah'
+    'notty' 'ah'
   )
   # Commands that would need to skip one positional argument:
   #    flock
